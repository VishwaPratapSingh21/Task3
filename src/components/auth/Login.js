import React, {Component} from 'react';
import axiosInstance from '../Apicalls';
import {notify} from 'react-notify-toast';
import {Link, Redirect} from 'react-router-dom';
import PasswordResetEmail from './ResetEmail';
import PropTypes from 'prop-types'

/**
 * Component for handling login.
 * @author [Hadijah kyampeire](https://github.com/hadijahkyampeire/Yummy_Reactjs_frontend)
 */
class Login extends Component {
    // initial states
    state = {
        password_field: '',
        email_field: '',
        error:null,
    }

    static propTypes = {
        login: PropTypes.func.isRequired
    }
    // handles changes and updates state
    handleInputChange = (event) => {
        const {name, value} = event.target;
        this.setState({[name]: value});
    }
    // function to make an API call on submit
    handleLogin = (event) => {
        const {email_field: email, password_field: password} = this.state;

        event.preventDefault();
        axiosInstance
            .post('auth/login', {email, password})
            .then(response => {
                localStorage.setItem('accessToken', response.data.access_token);
                localStorage.setItem('userEmail', response.data.user_email);
                localStorage.setItem('username', response.data.username);
                localStorage.setItem('identity', response.data.Id);
                this
                    .props
                    .login();
                notify.show(response.data.message, 'success', 4000);
            })
            // error handling
            .catch(error => {
                if (error.response) {
                    this.setState({error: error.response.data.message})
                    // notify.show(error.response.data.message, 'error', 3000)
                } else if (error.request) {
                    notify.show("Request not made", 'error', 3000)
                }
            });
    }
    render() {
        const {password_field, email_field} = this.state;
        const {from} = this.props.location.state || {
            from: {
                pathname: '/dashboard'
            }
        }

        if (this.props.loggedIn) {
            return (<Redirect to={from}/>)
        }
        
        return (
            <div className="background">
                <div id="logbar">
                    <div className="form-group">
                    <i className="fa fa-user-circle fa-5x" style={{color:"#26A69A"}}></i>
                        <div>
                            LOGIN
                        </div>
                        <form onSubmit={this.handleLogin}>
                        {this.state.error ? (
                    <div className="alert alert-danger">{this.state.error}
                     <button
                    type="button"
                    class="close"
                    data-dismiss="alert"
                    aria-label="Close"
                  >
                    <span aria-hidden="true">&times;</span>
                  </button></div>
                  ) : (
                    ""
                  )}
                            <div className="input-group">
                                <div className="input-group-addon pr-4" ><i className='fa fa-envelope'/></div>
                                <input
                                    type="email"
                                    id="emailsignup"
                                    name="email_field"
                                    aria-describedby="email"
                                    className="form-control"
                                    placeholder="example@gmail.com"
                                    onChange={this.handleInputChange}
                                    value={email_field}/>
                            </div>
                            <br/>
                            <div className="input-group">
                                <div className="input-group-addon pr-4" ><i className='fa fa-lock'/></div>
                                <input
                                    type="password"
                                    name="password_field"
                                    className="form-control"
                                    onChange={this.handleInputChange}
                                    value={password_field}
                                    placeholder="Password"/>
                            </div>
                            <br/>

                            <button name="signin" className="btn btn-default btn-lg" id="signin" type="submit">
                               
                                    LOGIN
                            </button>
                        </form>
                        <div className="change_link">
                            Not a member yet ?
                            <Link to="/register">
                                < h4 className="toregister">Register Here</h4>
                            </Link>
                        </div>
                    </div>
                    <a href="#" data-toggle="modal" data-target="#forgot_password">ForgotPassword?</a>
                    <PasswordResetEmail/>
                </div>
                <footer className="footer">
                    <p>&copy; Hadijah 2018</p>
                </footer>
            </div>
        );
    }
}

export default Login;
