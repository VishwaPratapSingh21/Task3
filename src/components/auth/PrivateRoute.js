import React from 'react';
import { Route, Redirect } from 'react-router-dom';

// function for private urls which cant be accessed before login
const PrivateRoute = ({ component: Component, ...rest }) => (
  <Route
    {...rest}
    render={
                props => (
                    rest.loggedIn ?
                    (<Component {...props} {...rest} />)
                    :
                    (<Redirect to={{
                        pathname: '/login',
                        state: { from: props.location },
                    }}
                    />)
                )
            }
  />
);
export default PrivateRoute;
