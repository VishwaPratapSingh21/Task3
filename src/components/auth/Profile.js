import React from 'react';

const Profile = () => (
    <div className="modal fade" id="userprofile" tabIndex="-1" role="dialog"
        aria-labelledby="exampleModalLabel" aria-hidden="true">
            <div className="modal-dialog" role="document">
    <div className="modal-content">
      <div className="modal-body">
      <div className="card"  >
  <div className="card-body text-center">
  <i className="fa fa-address-card fa-5x"/>
    <ul><h4 className="card-title ">User details</h4></ul>
    <p className="card-text"><strong>Email</strong>: {localStorage.getItem('userEmail')}</p>
    <p className="card-text"><strong>Username</strong>: {localStorage.getItem('username')}</p>
    <p className="card-text"><strong>Identity</strong>: {localStorage.getItem('identity')}</p>
  </div>
</div>
      </div>
      <div className="modal-footer">
        <button type="button" className="btn btn-secondary" data-dismiss="modal">Back</button>
      </div>
    </div>
  </div>
</div>
    
);
export default Profile;

