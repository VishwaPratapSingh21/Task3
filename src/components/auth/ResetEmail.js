import React, { Component } from 'react';
import {notify} from 'react-notify-toast';
import axiosInstance from '../Apicalls';

/**
 * Component for handling emails when resetting password.
 * @author [Hadijah kyampeire](https://github.com/hadijahkyampeire/Yummy_Reactjs_frontend)
 */
class PasswordResetEmail extends Component {
    state = {
        email:'',
        error:null,
    }

    handleEmail =(event) =>{
        const {name, value} = event.target;
        this.setState({[name]:value})
    }

    handleResetEmail =(event) =>{
        const {email} =this.state;
        event.preventDefault();
        axiosInstance.post(`auth/send_email`, {email})
        .then(response =>{
            document.getElementById("close-email-modal").click()
            notify.show(response.data.message, 'success', 4000);
        }).catch(error=>{
          if(error.response){
            this.setState({error: error.response.data.message})
            // notify.show(error.response.data.message, 'error', 4000)
          }else if(error.request){
            notify.show("Request not made", 'error', 3000)
          }

        })

    }

  render() {
    return (
      <div className="modal fade" id="forgot_password" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
        <div className="modal-dialog" role="document">
          <div className="modal-content">
            <div className="modal-header">
              <h5 className="modal-title">Reset password</h5>
              <button type="button" className="close" data-dismiss="modal" aria-label="Close" id="close-email-modal">
                <span aria-hidden="true">&times;</span>
              </button>
            </div>
            <form onSubmit={this.handleResetEmail}>
            <div className="modal-body">
            {this.state.error ? (
                    <div className="alert alert-danger">{this.state.error}
                     <button
                    type="button"
                    class="close"
                    data-dismiss="alert"
                    aria-label="Close"
                  >
                    <span aria-hidden="true">&times;</span>
                  </button></div>
                  ) : (
                    ""
                  )}
              <input
                type="email"
                name="email"
                onChange={this.handleEmail}
                value={this.state.email}
                aria-describedby="email"
                className="form-control"
                placeholder="example@gmail.com"
              />
            </div>
            <div className="modal-footer">
              <button type="submit" id="sendemail" className="btn btn-primary">Send Email</button>
              <button type="button" className="btn btn-secondary" data-dismiss="modal">Cancel</button>

            </div>
            </form>
          </div>
        </div>
      </div>
    );
  }
}
export default PasswordResetEmail;
