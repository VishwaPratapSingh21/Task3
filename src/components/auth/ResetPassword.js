import React, { Component } from 'react';
import {notify} from 'react-notify-toast';
import  axiosInstance from "../Apicalls";

/**
 * Component for resetting password.
 * @author [Hadijah kyampeire](https://github.com/hadijahkyampeire/Yummy_Reactjs_frontend)
 */
class ResetPassword extends Component {
    state = {
        email:'',
        password:'',
        retyped_password:'',
        token: null,

    }
    // called when the component has been rendered
    componentDidMount = () => {
        const search = this.props.location.search;
        const token = decodeURIComponent(search.substring(4, search.length));
        this.setState({ token })
    };

    handleInputChange =(event) =>{
        const {name, value } =event.target;
        this.setState({[name]:value});
    }

    handleResetPassword = (e)=>{
      // prevent the default action of the event to be triggered
        e.preventDefault();
        const headers = {Authorization: `Bearer ${this.state.token}`}
        const {email, password, retyped_password} =this.state; 
        axiosInstance.put(`auth/reset_password`, {email, password, retyped_password},{headers})
        .then(response=>{
            // success message once the Api call is successful
            notify.show(response.data.message, 'success', 4000)
            this.props.history.push('/login')
        }).catch(error=>{
            if (error.response){
              // notify about errors due to response
                notify.show(error.response.data.message, 'error', 3000)
            }else if(error.request){
              // error message due to request
                notify.show("Request not made", 'error', 3000)
            }
        })
    }
    
  render() {
    return (
      <div className="background">
      <div id="changepassword">

      <div id="form-group">
      <h3>Password Reset</h3>
        <form onSubmit={this.handleResetPassword}>
          <div className="input-group">
            <span className="input-group-addon pr-4" id="email"><i className="fa fa-envelope" /></span>
            <input
              type="email"
              id="emailsignup"
              name="email"
              value={this.state.email}
              onChange={this.handleInputChange}
              aria-describedby="email"
              className="form-control"
              placeholder="example@gmail.com"
            />
          </div>
          <br />
          <div className="input-group">
            <span className="input-group-addon pr-4" id="email"><i className="fa fa-lock" /></span>
            <input
              type="password"
              name="password"
              className="form-control"
              value={this.state.password}
              onChange={this.handleInputChange}
              placeholder="Password"
            />
          </div>
          <br />
          <div className="input-group">
            <span className="input-group-addon pr-4" id="email"><i className="fa fa-lock" /></span>
            <input
              type="password"
              name="retyped_password"
              className="form-control"
              value={this.state.retyped_password}
              onChange={this.handleInputChange}
              placeholder="retype_Password"
            />
          </div>
          <br />

          <button name="signin" className="btn btn-default" id="signin" type="submit">
            <i className="glyphicon glyphicon-log-in"> Save</i>
          </button>
        </form>
        </div>
        </div>
      </div>

    );
  }
}
export default ResetPassword;
