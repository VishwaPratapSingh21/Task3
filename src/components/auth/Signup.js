import React, {Component} from 'react';
import axiosInstance from '../Apicalls';
import {notify} from 'react-notify-toast'
import {Link, Redirect} from 'react-router-dom';
import {Nav} from './Landing'

class Signup extends Component {
    state = {
        password_field: '',
        email_field: '',
        username: '',
        error:null,
    }

    handleInputChange = (event) => {
        const {name, value} = event.target;
        this.setState({[name]: value});
    }

    handleRegister = (event) => {
        const {email_field: email, password_field: password, username: username} = this.state;

        event.preventDefault();
        axiosInstance
            .post('auth/register', {username, email, password})
            .then(response => {
                notify.show(response.data.message, 'success', 4000);
                this
                    .props
                    .history
                    .push('/login');
            })
            .catch(error => {
                if (error.response) {
                    this.setState({error: error.response.data.message})
                    // notify.show(error.response.data.message, 'error', 3000)
                } else if (error.request) {
                    notify.show("Request not made", 'error', 3000)
                }
            });
    }

    render() {
        const {username, password_field, email_field} = this.state;
        let user = this.state.username
        
        if(this.props.loggedIn){
            // if one is loggedin it should not show signup anymore
            return(<Redirect to='/dashboard'/>)
        }

        return (
            
            <div className="background">
                <div id="signbar">
                        <div >
                    <i className="fa fa-user-circle fa-5x" style={{color:"#26A69A"}}></i>
                    <div> REGISTER </div>
                </div>
                    <form onSubmit={this.handleRegister}>
                    {this.state.error ? (
                    <div className="alert alert-danger">{this.state.error}
                     <button
                    type="button"
                    class="close"
                    data-dismiss="alert"
                    aria-label="Close"
                  >
                    <span aria-hidden="true">&times;</span>
                  </button></div>
                  ) : (
                    ""
                  )}
                        <div className="input-group">
                            <span className="input-group-addon pr-4" id="username"><i className='fa fa-user'/></span>
                            <input
                                id="username"
                                name="username"
                                aria-describedby="username"
                                className="form-control"
                                placeholder="example"
                                onChange={this.handleInputChange}
                                value={username}/>
                        </div>
                        <br/>
                        <div className="input-group">
                            <span className="input-group-addon pr-4" id="email"><i className='fa fa-envelope'/></span>
                            <input
                                id="emailsignup"
                                name="email_field"
                                aria-describedby="email"
                                className="form-control"
                                placeholder="example@gmail.com"
                                onChange={this.handleInputChange}
                                value={email_field}/>
                        </div>
                        <br/>
                        <div className="input-group">
                            <span className="input-group-addon pr-4" id="email"><i className='fa fa-lock'/></span>
                            <input
                                type="password"
                                name="password_field"
                                className="form-control"
                                onChange={this.handleInputChange}
                                value={password_field}
                                placeholder="Password"/>
                        </div>
                        <br/>
                            <div className="text-center">
                                <input name="signup" id="signup" value="Signup" className="btn btn-default btn-lg" type="submit"/>
                            
                        </div>

                        <div className="change_link">
                            Already a member ?
                            <Link to="/login"><h4 className="toregister">Login here</h4>
                            </Link>
                        </div>

                    </form>
                </div>
            </div>
        );
    }
}
export default Signup;
