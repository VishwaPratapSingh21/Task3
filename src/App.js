import React, {Component} from 'react';
import {Switch, Route, BrowserRouter, Link} from 'react-router-dom';
import Notifications from 'react-notify-toast';
import {notify} from 'react-notify-toast';

import Signup from './components/auth/Signup';
import ResetPassword from './components/auth/ResetPassword';
import Login from './components/auth/Login';
import About from './components/auth/About';
import Categories from './components/categories/Dashboard';
import ViewRecipes from './components/recipes/ViewRecipes';
import PrivateRoute from './components/auth/PrivateRoute';
import Landing, {Nav} from './components/auth/Landing';

import axiosInstance from './components/Apicalls';

const NotFound = () => (
  <div className="container">
    <div className="row">
        <div className="col-md-12">
            <div className="error-template">
                <h1>
                    Oops!</h1>
                <h2>
                    Page Not Found, 404</h2>
                <div className="error-details">
                    Sorry, an error has occured, Requested page can not be found!
                </div>
                <div className="error-actions">
                    <Link to="/" className="btn btn-primary btn-lg" style={{backgroundColor:'#009688'}}><span className="fa fa-home"></span>
                        Take Me Home </Link>
                </div>
            </div>
        </div>
    </div>
</div>
);

class App extends Component {
  state = {
    loggedin: false
  }
  /**
   * @param {Object} event - 
   */
  logoutUser = (event) => {
    event.preventDefault();
    axiosInstance
      .post(`auth/logout`,null)
      .then(response => {
        this.setState({loggedin: false})
        localStorage.clear();
        notify.show(response.data.message, 'success', 4000);
      })
      .catch(e => {
      })
  }

  loginUser = ()=>{
    this.setState({
      loggedin: true
    })
  }

  componentDidMount(){
    if(localStorage.getItem("accessToken")){
      this.loginUser();
    }
  }

  render() {
    return (
      <BrowserRouter>
        <div>
          <Nav logout={this.logoutUser} loggedIn={this.state.loggedin}/>
          <Notifications/>
          <Switch>
            <Route exact path="/" component={props=>(<Landing {...props} loggedIn={this.state.loggedin}/>)}/>
            <Route exact path="/register" component={props=>(<Signup {...props} loggedIn={this.state.loggedin}/>)}/>
            <Route exact path="/reset" component={ResetPassword}/>
            <PrivateRoute exact path="/about" component={About} loggedIn={this.state.loggedin} />
            <Route exact path="/login" component={props => (<Login {...props} login={this.loginUser} loggedIn={this.state.loggedin} />)}/>
            <PrivateRoute exact path="/dashboard" component={Categories}  loggedIn={this.state.loggedin} />
            <PrivateRoute exact path="/category/:id/recipes" component={ViewRecipes} loggedIn={this.state.loggedin}/>
            <Route component={NotFound}/>
          </Switch>
        </div>
      </BrowserRouter>
    );
  }
}

export default App;
