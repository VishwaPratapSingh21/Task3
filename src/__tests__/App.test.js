import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import toJson, { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import App from '../App';
import { Nav } from '../components/auth/Landing';
import PrivateRoute from '../components/auth/PrivateRoute';

const constantDate = new Date('2018-02-13T12:41:20');

Date = class extends Date {
  constructor() {
    return constantDate;
  }
};

global.Date = Date;

describe('App component', () => {
  const wrapper = shallow(<App />);

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('has the the expected components', () => {
    expect(wrapper.find('BrowserRouter')).toHaveLength(1);
    expect(wrapper.find('div').length).toBe(1);
  });

  it('renders a Nav component in the div', () => {
    expect(wrapper.find('div').props().children.find(Nav)).toBeTruthy();
  });

  it('renders the routes', () => {
    expect(wrapper.find('Route').length).toEqual(5);
    expect(wrapper.find('PrivateRoute').length).toEqual(3);
  });

  it('renders the switch', () => {
    expect(wrapper.find('Switch')).toHaveLength(1);
  });

  it('renders the notifications wrapper', () => {
    expect(wrapper.find('_class')).toHaveLength(1);
  });

  it('has the right initial state', () => {
    expect(wrapper.state().loggedin).toBe(false);
  });

  it('calls component did mount', () => {
    sinon.spy(App.prototype, 'componentDidMount');
    const fullWrapper = mount(<App />);
    expect(App.prototype.componentDidMount).toHaveProperty('callCount', 1);
    App.prototype.componentDidMount.restore();
  });

  it('fully renders properly', () => {
    const fullWrapper = mount(<App />);
    expect(toJson(fullWrapper)).toMatchSnapshot();
    expect(fullWrapper.find('Landing')).toHaveLength(1);
  });
});
