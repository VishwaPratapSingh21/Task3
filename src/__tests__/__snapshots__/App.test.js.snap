// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component fully renders properly 1`] = `
<App>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div>
        <Component
          loggedIn={false}
          logout={[Function]}
        >
          <nav
            className="navbar navbar-toggleable-md navbar-light"
            style={
              Object {
                "backgroundColor": "#009688",
                "borderRadius": 0,
                "marginBottom": 0,
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <button
              aria-controls="navbarTogglerDemo02"
              aria-expanded="false"
              aria-label="Toggle navigation"
              className="navbar-toggler navbar-toggler-right"
              data-target="#navbarTogglerDemo02"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>
            <a
              className="navbar-brand"
              href="#"
            >
              <div
                className="logo"
              />
              Recipes by KH
            </a>
            <div
              className="collapse navbar-collapse"
              id="navbarTogglerDemo02"
              style={
                Object {
                  "marginBottom": 0,
                }
              }
            >
              <ul
                className="navbar-nav ml-auto"
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                <li
                  className="nav-item"
                >
                  <NavLink
                    activeClassName="active"
                    activeStyle={
                      Object {
                        "borderBottomColor": "white",
                        "color": "white",
                      }
                    }
                    ariaCurrent="true"
                    className="nav-link"
                    exact={true}
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                    to="/"
                  >
                    <Route
                      exact={true}
                      path="/"
                    >
                      <Link
                        aria-current="true"
                        className="nav-link active"
                        replace={false}
                        style={
                          Object {
                            "borderBottomColor": "white",
                            "color": "white",
                          }
                        }
                        to="/"
                      >
                        <a
                          aria-current="true"
                          className="nav-link active"
                          href="/"
                          onClick={[Function]}
                          style={
                            Object {
                              "borderBottomColor": "white",
                              "color": "white",
                            }
                          }
                        >
                          <i
                            className="fa fa-home "
                          />
                           
                           
                          Home
                        </a>
                      </Link>
                    </Route>
                  </NavLink>
                </li>
                <li
                  className="nav-item"
                >
                  <NavLink
                    activeClassName="active"
                    activeStyle={
                      Object {
                        "borderBottomColor": "white",
                        "color": "white",
                      }
                    }
                    ariaCurrent="true"
                    className="nav-link"
                    id="signinh"
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                    to="/login"
                  >
                    <Route
                      path="/login"
                    >
                      <Link
                        aria-current={false}
                        className="nav-link"
                        id="signinh"
                        replace={false}
                        style={
                          Object {
                            "color": "white",
                          }
                        }
                        to="/login"
                      >
                        <a
                          aria-current={false}
                          className="nav-link"
                          href="/login"
                          id="signinh"
                          onClick={[Function]}
                          style={
                            Object {
                              "color": "white",
                            }
                          }
                        >
                          <span>
                            <i
                              className="fa fa-sign-in "
                            />
                             Login
                          </span>
                        </a>
                      </Link>
                    </Route>
                  </NavLink>
                </li>
                <li
                  className="nav-item"
                >
                  <NavLink
                    activeClassName="active"
                    activeStyle={
                      Object {
                        "borderBottomColor": "white",
                        "color": "white",
                      }
                    }
                    ariaCurrent="true"
                    className="nav-link"
                    id="signupj"
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                    to="/register"
                  >
                    <Route
                      path="/register"
                    >
                      <Link
                        aria-current={false}
                        className="nav-link"
                        id="signupj"
                        replace={false}
                        style={
                          Object {
                            "color": "white",
                          }
                        }
                        to="/register"
                      >
                        <a
                          aria-current={false}
                          className="nav-link"
                          href="/register"
                          id="signupj"
                          onClick={[Function]}
                          style={
                            Object {
                              "color": "white",
                            }
                          }
                        >
                          <i
                            className="fa fa-user-plus "
                          />
                           
                           
                          Signup
                        </a>
                      </Link>
                    </Route>
                  </NavLink>
                </li>
              </ul>
            </div>
            <Profile>
              <div
                aria-hidden="true"
                aria-labelledby="exampleModalLabel"
                className="modal fade"
                id="userprofile"
                role="dialog"
                tabIndex="-1"
              >
                <div
                  className="modal-dialog"
                  role="document"
                >
                  <div
                    className="modal-content"
                  >
                    <div
                      className="modal-body"
                    >
                      <div
                        className="card"
                      >
                        <div
                          className="card-body text-center"
                        >
                          <i
                            className="fa fa-address-card fa-5x"
                          />
                          <ul>
                            <h4
                              className="card-title "
                            >
                              User details
                            </h4>
                          </ul>
                          <p
                            className="card-text"
                          >
                            <strong>
                              Email
                            </strong>
                            : 
                          </p>
                          <p
                            className="card-text"
                          >
                            <strong>
                              Username
                            </strong>
                            : 
                          </p>
                          <p
                            className="card-text"
                          >
                            <strong>
                              Identity
                            </strong>
                            : 
                          </p>
                        </div>
                      </div>
                    </div>
                    <div
                      className="modal-footer"
                    >
                      <button
                        className="btn btn-secondary"
                        data-dismiss="modal"
                        type="button"
                      >
                        Back
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </Profile>
          </nav>
        </Component>
        <_class
          options={Object {}}
        >
          <div
            id="notification-wrapper"
          />
        </_class>
        <Switch>
          <Route
            component={[Function]}
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            path="/"
          >
            <component
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <Landing
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                loggedIn={false}
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <Slideshow>
                  <div
                    className="container-fluid"
                  >
                    <div
                      className="carousel slide"
                      data-ride="carousel"
                      id="myCarousel"
                    >
                      <ol
                        className="carousel-indicators"
                      >
                        <li
                          className="active"
                          data-slide-to="0"
                          data-target="#myCarousel"
                        />
                        <li
                          data-slide-to="1"
                          data-target="#myCarousel"
                        />
                        <li
                          data-slide-to="2"
                          data-target="#myCarousel"
                        />
                      </ol>
                      <div
                        className="carousel-inner"
                        role="listbox"
                        style={
                          Object {
                            "height": "100%",
                          }
                        }
                      >
                        <FoodImages
                          image="food4.jpg"
                          status="active"
                          text="family meal"
                        >
                          <div
                            className="carousel-item active"
                            style={
                              Object {
                                "height": -50,
                              }
                            }
                          >
                            <img
                              alt="family meal"
                              className="d-block img-fluid"
                              src="food4.jpg"
                              style={
                                Object {
                                  "backgroundPosition": "cover",
                                  "height": "100%",
                                  "width": "100%",
                                }
                              }
                            />
                            <div
                              className="carousel-caption"
                              style={
                                Object {
                                  "padding": 0,
                                }
                              }
                            >
                              <div
                                id="welcome"
                              >
                                <h3>
                                  YOU LOVE COOKING?
                                </h3>
                                <h3>
                                  <span>
                                    WELCOME TO YUMMY RECIPES
                                  </span>
                                </h3>
                                <h4>
                                  SIGNUP AND GET STARTED
                                </h4>
                                <div>
                                  <h2>
                                    It is
                                    <Clock>
                                      <div
                                        className="clock"
                                      >
                                        12
                                        :
                                        41
                                        :
                                        20
                                         
                                        pm
                                      </div>
                                    </Clock>
                                  </h2>
                                </div>
                              </div>
                            </div>
                          </div>
                        </FoodImages>
                        <FoodImages
                          image="new1.jpg"
                          status=""
                          text="proteins one"
                        >
                          <div
                            className="carousel-item "
                            style={
                              Object {
                                "height": -50,
                              }
                            }
                          >
                            <img
                              alt="proteins one"
                              className="d-block img-fluid"
                              src="new1.jpg"
                              style={
                                Object {
                                  "backgroundPosition": "cover",
                                  "height": "100%",
                                  "width": "100%",
                                }
                              }
                            />
                            <div
                              className="carousel-caption"
                              style={
                                Object {
                                  "padding": 0,
                                }
                              }
                            >
                              <div
                                id="welcome"
                              >
                                <h3>
                                  YOU LOVE COOKING?
                                </h3>
                                <h3>
                                  <span>
                                    WELCOME TO YUMMY RECIPES
                                  </span>
                                </h3>
                                <h4>
                                  SIGNUP AND GET STARTED
                                </h4>
                                <div>
                                  <h2>
                                    It is
                                    <Clock>
                                      <div
                                        className="clock"
                                      >
                                        12
                                        :
                                        41
                                        :
                                        20
                                         
                                        pm
                                      </div>
                                    </Clock>
                                  </h2>
                                </div>
                              </div>
                            </div>
                          </div>
                        </FoodImages>
                        <FoodImages
                          image="new2.jpg"
                          status=""
                          text="proteins"
                        >
                          <div
                            className="carousel-item "
                            style={
                              Object {
                                "height": -50,
                              }
                            }
                          >
                            <img
                              alt="proteins"
                              className="d-block img-fluid"
                              src="new2.jpg"
                              style={
                                Object {
                                  "backgroundPosition": "cover",
                                  "height": "100%",
                                  "width": "100%",
                                }
                              }
                            />
                            <div
                              className="carousel-caption"
                              style={
                                Object {
                                  "padding": 0,
                                }
                              }
                            >
                              <div
                                id="welcome"
                              >
                                <h3>
                                  YOU LOVE COOKING?
                                </h3>
                                <h3>
                                  <span>
                                    WELCOME TO YUMMY RECIPES
                                  </span>
                                </h3>
                                <h4>
                                  SIGNUP AND GET STARTED
                                </h4>
                                <div>
                                  <h2>
                                    It is
                                    <Clock>
                                      <div
                                        className="clock"
                                      >
                                        12
                                        :
                                        41
                                        :
                                        20
                                         
                                        pm
                                      </div>
                                    </Clock>
                                  </h2>
                                </div>
                              </div>
                            </div>
                          </div>
                        </FoodImages>
                      </div>
                      <a
                        className="carousel-control-prev"
                        data-slide="prev"
                        href="#myCarousel"
                        role="button"
                      >
                        <span
                          aria-hidden="true"
                          className="carousel-control-prev-icon"
                        />
                        <span
                          className="sr-only"
                        >
                          Previous
                        </span>
                      </a>
                      <a
                        className="carousel-control-next"
                        data-slide="next"
                        href="#myCarousel"
                        role="button"
                      >
                        <span
                          aria-hidden="true"
                          className="carousel-control-next-icon"
                        />
                        <span
                          className="sr-only"
                        >
                          Next
                        </span>
                      </a>
                    </div>
                  </div>
                </Slideshow>
              </Landing>
            </component>
          </Route>
        </Switch>
      </div>
    </Router>
  </BrowserRouter>
</App>
`;

exports[`App component renders properly 1`] = `
<BrowserRouter>
  <div>
    <Component
      loggedIn={false}
      logout={[Function]}
    />
    <_class
      options={Object {}}
    />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/register"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/reset"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        loggedIn={false}
        path="/about"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/login"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        loggedIn={false}
        path="/dashboard"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        loggedIn={false}
        path="/category/:id/recipes"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</BrowserRouter>
`;
